junit-platform-engine/src/main/java/org/junit/platform/engine/support/descriptor/UriSource.java

```diff
@@ -33,4 +39,42 @@
 	 */
 	URI getUri();
 
+	/**
+	 * Create a new {@code UriSource} using the supplied {@code uri}.
+	 * <p>
+	 * This implementation tries resolve the {@code uri} to local file
+	 * system path-based source first. If that fails for any reason, an
+	 * instance of a simple default uri source class storing the supplied
```

Comment:
an instance of a simple default uri source class --> an instance of the default {@code UriSource} implementation